### DISCUSSION TITLE
Base64 rewrite btoa and atob

### DISCUSSION THREAD

**Discussion by sudazzle** [mdn/mdn-community#619] [Content] [answered]
This is a question regarding a rewrite section of btoa and atob
https://developer.mozilla.org/en-US/docs/Glossary/Base64#solution_2_%E2%80%93_rewriting_atob_and_btoa_using_typedarrays_and_utf-8

```
for (var nMapIdx = 0; nMapIdx < nStrLen; nMapIdx++) {
    nChr = sDOMStr.charCodeAt(nMapIdx);
    nArrLen += nChr < 0x80 ? 1 : nChr < 0x800 ? 2 : nChr < 0x10000 ? 3 : nChr < 0x200000 ? 4 : nChr < 0x4000000 ? 5 : 6;
}
```
In the code section above (nside a function to convert binary string to UTF-8 (strToUTF8Arr)) charCodeAt return value between (0 and 65535) as of (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/charCodeAt). Should it be using codePointAt?

**Comment by sudazzle** [comment]
I have created an issue related to this one.
https://github.com/mdn/content/issues/11400

**Comment by AnilSeervi** [comment, marked as answer]
- Fixed in #11826

**Comment by pepelsbey** [comment]
Closing as answered/settled.
